src/pybind/mgr/dashboard/controllers/__init__.py

```diff
@@ -1105,3 +1105,11 @@ def validate_args(*args, **kwargs):
             return func(*args, **kwargs)
         return validate_args
     return decorator
+
+def merge_list_of_dicts_by_key(target_list: list, source_list: list, key: str):
```

Comment:
This should be placed in `src/pybind/mgr/dashboard/tools.py`.

Can you add tests for this method to ensure it works properly?